jsonschema/src/keywords/one_of.rs

```diff
@@ -7,12 +7,15 @@ use crate::{
     primitive_type::PrimitiveType,
     schema_node::SchemaNode,
     validator::{format_iter_of_validators, PartialApplication, Validate},
+    schemas::Discriminator,
 };
 use serde_json::{Map, Value};
+use std::collections::HashMap;
 
 pub(crate) struct OneOfValidator {
-    schemas: Vec<SchemaNode>,
+    schemas: HashMap<String, SchemaNode>,
```

Comment:
Could be possible to leave this as is and try and match the discriminator against the SchemaNode `relative_path` this would reduce the change set a bit.